lib_audio_dsp/api/dsp/biquad.h

```diff
@@ -21,26 +21,69 @@ int32_t adsp_biquad(
   int32_t state[8],
   left_shift_t lsh);
 
+/**
+ * @brief Slewing biquad state structure
+ */
+typedef struct {
+  /** Target filter coefficients, the active coefficients are slewed towards these */
+  q2_30 DWORD_ALIGNED target_coeffs[8];
+  /** Active filter coefficients, used to filter the audio */
+  q2_30 DWORD_ALIGNED active_coeffs[8];
+  /** Left shift compensation for if the filter coefficents cannot be represented in Q1.30 */
+  left_shift_t lsh;
+  /** Shift value used by the exponential slew */
+  int32_t slew_shift;
+  /** Remaining shifts for cases when the left shift changes during a target_coeff update. */
+  left_shift_t remaining_shifts;
+
+} biquad_slew_t;
 
 /**
- * @brief Biquad filter with slew.
- *  This function implements a biquad filter with slew. 
- *  The filter is implemented as a direct form 1.
- *  The coeffs are exponentially slewed towards the target_coeffs.
- *  This is can be used for real time adjustable biquads.
+ * @brief Initialise a slewing biquad filter object.
+ *  This sets the active filter coefficients to the target value
  * 
- * @param new_sample      New sample to be filtered
- * @param coeffs          Filter coefficients
- * @param target_coeffs   Target filter coefficients
- * @param state           Filter state
- * @param lsh             Left shift compensation value
+ * @param target_coeffs   Filter coefficients
+ * @param lsh             Filter left shift compensation value
  * @param slew_shift      Shift value used in the exponential slew
- * @return int32_t        Filtered sample
+ * @return biquad_slew_t  Slewing biquad object
  */
-int32_t adsp_biquad_slew(
-  int32_t new_sample,
-  q2_30 coeffs[8],
+biquad_slew_t adsp_biquad_slew_init(
   q2_30 target_coeffs[8],
-  int32_t state[8],
   left_shift_t lsh,
-  int32_t slew_shift);
\ No newline at end of file
+  left_shift_t slew_shift
+);
+
+/**
+ * @brief Update the target coefficients in a slewing biquad filter object.
+ *  This updates the target coefficients, and manages any change in filter
+ *  coefficient left shift. This may require shifting the active filter
+ *  coefficients and states.
+ * 
+ * @param slew_state       Slewing biquad state object
+ * @param states           Filter state for each biquad channel
+ * @param channels         Number of channels in states
+ * @param target_coeffs    New filter coefficients
+ * @param lsh              New filter left shift compensation value
+ */
+void adsp_biquad_slew_update(
```

Comment:
Sorry to ask you again, but could you make this one adsp_biquad_slew_update_coeffs?
This one seems a bit confusing